src/app/pages/details/details.js

```diff
@@ -107,11 +108,16 @@ function AnotherLanguage({locale, translation}) {
 function LinksToTranslations() {
     const {translations: [translations=[]], meta: {locale}} = useDetailsContext();
     const localLanguage = languageTranslations[locale][locale];
+    const {language, setLanguage} = useLanguageContext();
 
     if (translations.length === 0) {
         return null;
     }
 
+    if (language !== locale) {
+        setLanguage(locale);
+    }
```

Comment:
this is modifying state during render.  Should it be enclosed inside a useEffect?
```
useEffect(() => {
  if (language !== locale) {
         setLanguage(locale);
   }
}, [locale, language])
```